internal-dns/types/src/config.rs

```diff
@@ -533,8 +552,68 @@ impl DnsConfigBuilder {
         )
     }
 
-    /// Construct a `DnsConfigZone` describing the control plane zone described
-    /// up to this point
+    /// Higher-level shorthand for adding an internal DNS zone, including
+    /// records for both its HTTP and DNS interfaces.
+    ///
+    /// # Errors
+    ///
+    /// This fails if the provided `service` is not for an internal DNS zone. It
+    /// also fails if the given zone has already been added to the
+    /// configuration.
+    pub fn host_zone_internal_dns(
+        &mut self,
+        zone_id: OmicronZoneUuid,
+        service: ServiceName,
+        http_address: SocketAddrV6,
+        dns_address: SocketAddr,
+    ) -> anyhow::Result<()> {
+        anyhow::ensure!(
+            service == ServiceName::InternalDns,
+            "This method is only valid for internal DNS servers, \
+            but we were provided the service '{service:?}'",
+        );
```

Comment:
Would it make sense to just remove this argument?

Reply:
i think so. i kept it here for consistency with `host_zone_clickhouse_{single_node,cluster,keeper}()`, but i think it should be removed from them too. removing the argument from all these helpers is one of the more immediate followup changes i'd like to make.